consul/src/main/java/com/linecorp/armeria/internal/consul/AgentServiceClient.java

```diff
@@ -0,0 +1,220 @@
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.MoreObjects;
+
+/**
+ * Consul API client to register, deregister service and get list of services.
+ * {@code AgentServiceClient} is responsible for endpoint of Consul API:
+ * {@code `/agent/service`}(https://www.consul.io/api/agent/service.html)
+ *
+ * <p>GET /agent/services
+ * GET /agent/service/:service_id
+ * PUT /agent/service/register
+ * PUT /agent/service/deregister/:service_id
+ */
+public final class AgentServiceClient {
+
+    private final ConsulClient client;
+
+    /**
+     * Builds an AgentServiceClient with a ConsulClient.
+     */
+    public static AgentServiceClient of(ConsulClient consulClient) {
+        return new AgentServiceClient(consulClient);
+    }
+
+    private AgentServiceClient(ConsulClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Gets a list of services from consul agent.
+     */
+    public CompletableFuture<Map<String, Service>> services(
+            @Nullable String serviceName, ConsulParameters parameters) {
+        String serviceFilter = "";
+        if (serviceName != null) {
+            serviceFilter = "filter=Service==" + serviceName + '&';
+        }
+        return client.consulWebClient()
+                     .get("/agent/services?" + serviceFilter + parameters.toParameterString())
+                     .aggregate()
+                     .thenApply(response -> {
+                         final ObjectMapper mapper = new ObjectMapper();
+                         try {
+                             return mapper.readValue(
+                                     response.content().toStringUtf8(),
+                                     mapper.getTypeFactory().constructMapType(
+                                             Map.class, String.class, Service.class));
+                         } catch (JsonProcessingException e) {
+                             throw new RuntimeException(e);
+                         }
+                     });
+    }
+
+    /**
+     * Registers a service.
+     */
+    public CompletableFuture<String> register(String serviceName, String address, int port,
+                                              ConsulParameters parameters) {
+        return register(serviceName, address, port, null, parameters);
+    }
+
+    /**
+     * Registers a service with a health checking endpoint.
+     *
+     * @return returns a generated service ID.
+     */
+    public CompletableFuture<String> register(String serviceName, String address, int port,
+                                              @Nullable Check check, ConsulParameters parameters) {
+        final String serviceId = UUID.randomUUID().toString();
```

Comment:
`UUID.randomeUUID()` internally uses `SecureRandom` which has a performance problem.
Please see #2224 what we did before. 

https://github.com/line/armeria/pull/2224/files#diff-387699884575f0fa0d25e5a74e7c32faR40
We can copy the code to create a random string.

Reply:
Any plan for this?